src/saveToCase.js

```diff
@@ -0,0 +1,575 @@
+
+            widget.getValue = function () {
+                var currentValues = {};
+                _.each(widget.input.find('.fd-index-property'), function (kvPair) {
+                    var $pair = $(kvPair);
+                    currentValues[$pair.find('.fd-index-property-name').val()] = {
+                        calculate: $pair.find('.fd-index-property-source').val(),
+                        case_type: $pair.find('.fd-index-property-case-type').val(),
+                        relationship: $pair.find('.fd-index-property-relationship').val(),
+                    };
+                });
+                return currentValues;
+            };
+
+            return widget;
+        };
+
+    var CASE_XMLNS = "http://commcarehq.org/case/transaction/v2",
+        INVALID_PROP_REGEX = /[a-z_]+/,
+        saveToCaseMugOptions = {
+            typeName: 'Save to Case',
+            isTypeChangeable: false,
+            isDataOnly: true,
+            supportsDataNodeRole: true,
+            icon: 'icon-save',
+            init: function (mug, form) {},
+            spec: {
+                xmlnsAttr: { presence: "optional" },
+                "date_modified": {
+                    lstring: "Date modified",
+                    visibility: 'visible',
+                    presence: 'required',
+                    widget: widgets.xPath,
+                },
+                "user_id": {
+                    lstring: "User ID",
+                    visibility: 'visible',
+                    presence: 'optional',
+                    widget: widgets.xPath
+                },
+                "case_id": {
+                    lstring: "Case ID",
+                    visibility: 'visible',
+                    presence: 'required',
+                    widget: widgets.xPath,
+                },
+                "use_create": {
+                    lstring: "Create Case",
+                    visibility: 'visible',
+                    presence: 'optional',
+                    widget: widgets.checkbox
+                },
+                "create_property": {
+                    lstring: "Properties To Create",
+                    visibility: 'visible',
+                    presence: 'optional',
+                    widget: saveCasePropWidget,
+                    validationFunc: function (mug) {
+                        if (mug.p.use_create) {
+                            var props = _.without(_.keys(mug.p.create_property), ""),
+                                required = ["case_type", "case_name"],
+                                optional = ["owner_id"],
+                                legal = _.union(required, optional),
+                                illegalProps = _.difference(props, legal),
+                                requiredProps = _.intersection(props, required),
+                                invalidProps = _.filter(props, function(p) {
+                                    return !INVALID_PROP_REGEX.test(p);
```

Comment:
The negatives here are making it hard for me to decipher the meaning of this code. Are these true statements?

- Any property that contains a lower-case letter or underscore is not valid.
- The set of invalid properties is all the properties that are (not not) valid. ... that can't be right.

Said another way, if I'm doing the logic right, this filter function returns true for any property that does not have at least one lower-case letter or underscore, and returning true means the property is not valid. So `"# %"` and `"AGE"` will return `true` (not valid), while `"age"` and `"#a b c"` will return `false`, meaning they are valid.

Suggestions:
- change the regex to `VALID_PROP_REGEX = /^[a-z_]+$/i`, and keep the `!` operator in the filter function.
- or `INVALID_PROP_REGEX = /[^a-z_]/i`, and drop the `!` operator from the filter function.

Note that I changed the regexps to case-insenstive since case properties are probably allowed to contain upper-case letters. Also note that these regexps do not allow unicode characters such as é, and I'm not sure if that matters or not. Please correct me if I got this all wrong.